src/backend/utils/description.ts

```diff
@@ -21,64 +21,73 @@ export abstract class Description {
+        case 'NgClass':
+          return p(Description._getClassDesc(instance));
+        case 'NgStyle':
+          return p(Description._getNgClassDesc(instance));
+        case 'NgFormModel':
+          return p(Description._getNgFormModelDesc(instance));
+        case 'NgFormControl':
+          return p(Description._getFormControlDesc(instance));
+        case 'NgControlStatus':
+          return p(Description._getControlStatusDesc(instance));
+        case 'NgModel':
+          return p(Description._getNgModelDesc(instance));
+        case 'NgForm':
+          return p(Description._getNgFormDesc(instance));
+        default:
+          return p([]);
+      }
+    } catch (e) {
+      return p([]);
     }
-    return p([]);
   }
 
   public static getComponentDescription(debugElement: any): Array<Property> {
     if (debugElement == null) {
       return [];
     }
 
-    const element: any = debugElement.nativeElement;
-
-    const componentName = debugElement.componentInstance
-      ? debugElement.componentInstance.constructor.name
-      : element.tagName.toLowerCase();
-
-    switch (componentName) {
-      case 'a':
-        return [
-          { key: 'text', value: element.text },
-          { key: 'url', value: element.hash }
-        ];
-      case 'NgSelectOption':
-        return Description._getSelectOptionDesc(element);
-      case 'NgIf':
-        return Description._getNgIfDesc(debugElement.componentInstance);
-      case 'NgControlName':
-        return Description._getControlNameDesc
+    try {
+      const element: any = debugElement.nativeElement;
+
+      const componentName = debugElement.componentInstance
+        ? debugElement.componentInstance.constructor.name
+        : element.tagName.toLowerCase();
+
+      switch (componentName) {
+        case 'a':
+          return [
+            {key: 'text', value: element.text},
+            {key: 'url', value: element.hash}
+          ];
+        case 'NgSelectOption':
+          return Description._getSelectOptionDesc(element);
+        case 'NgIf':
+          return Description._getNgIfDesc(debugElement.componentInstance);
+        case 'NgControlName':
+          return Description._getControlNameDesc
           (debugElement.componentInstance);
-      case 'NgSwitch':
-        return Description._getNgSwitchDesc(debugElement.componentInstance);
-      case 'NgSwitchWhen':
-        return Description._getNgSwitchWhenDesc
+        case 'NgSwitch':
+          return Description._getNgSwitchDesc(debugElement.componentInstance);
+        case 'NgSwitchWhen':
+          return Description._getNgSwitchWhenDesc
           (debugElement.componentInstance);
-      case 'NgSwitchDefault':
-        return Description._getNgSwitchWhenDesc
+        case 'NgSwitchDefault':
+          return Description._getNgSwitchWhenDesc
           (debugElement.componentInstance);
+        default:
+          return [];
+      }
+    } catch (e) {
+      return [];
```

Comment:
@tyllrb This means if one property read fails, that element will have no description properties? Could we easily change this so that only that property is missing in this case?